usaspending_api/references/v2/views/cfda.py

```diff
@@ -0,0 +1,48 @@
+from rest_framework.response import Response
+from rest_framework.views import APIView
+from requests import post
+from time import sleep
+from django.conf import settings
+
+ALL_CFDAS = None
+
+
+class CFDAViewSet(APIView):
+    """
+    Return an agency name and active fy.
+    """
+
+    endpoint_doc = "usaspending_api/api_contracts/contracts/v2/references/agency/id.md"
+
+    def get(self, request, cfda=None):
+        """
+        Return the view's queryset.
+        """
+        self._populate_cfdas_if_needed()
+
+        if cfda:
+            results = [result for result in ALL_CFDAS if result["cfda"] == cfda]
+        else:
+            results = ALL_CFDAS
+
+        response = {"results": results}
+
+        return Response(response)
+
+    def _populate_cfdas_if_needed(self):
+        global ALL_CFDAS
+        if not ALL_CFDAS:
+            response = self._request_from_grants_api()
+
+            #  grants API is brittle in practice, so if we don't get results retry at a polite rate
+            while not response:
+                sleep(2)
+                response = self._request_from_grants_api()
+
+            ALL_CFDAS = response.values()
```

Comment:
Wouldn't it be helpful to store the "cfdas" as the dictionary for looking up a specific CFDA? I would still be easy to  returning all CFDA records using `.values()`